File: examples/mer-verify.mjs
// Public synthetic fixture only. No network, wallet, private key or Axiom account.
import assert from "node:assert/strict";
import { createRequire } from "node:module";
import { readFileSync } from "node:fs";
import { verifyMer } from "@axiom-relay/mer";

const fixtures = JSON.parse(readFileSync(createRequire(import.meta.url).resolve("@axiom-relay/mer/vectors"), "utf8"));
const fixture = fixtures.vectors.find((item) => item.id === "legacy-fulfillment");
assert.ok(fixture, "Expected published fixture is missing");
const options = { trustedKeys: fixture.trustedKeys, verificationTime: fixture.verificationTime };
assert.equal(verifyMer(fixture.envelope, options).valid, true);
const changed = structuredClone(fixture.envelope);
changed.receipt.economics.totalAuthorized.amount = "0.02";
assert.equal(verifyMer(changed, options).valid, false);
console.log(JSON.stringify({ synthetic: true, originalVerified: true, tamperingRejected: true, networkRequests: 0 }));

File: examples/mer_verify.py
"""Public synthetic fixture only; no network, wallet, private key or Axiom account."""
import copy
import json
from importlib.resources import files
from axiom_mer import verify_mer

fixtures = json.loads(files("axiom_mer").joinpath("data/vectors.json").read_text())
fixture = next(item for item in fixtures["vectors"] if item["id"] == "legacy-fulfillment")
options = {"trusted_keys": fixture["trustedKeys"], "verification_time": fixture["verificationTime"]}
if not verify_mer(fixture["envelope"], **options)["valid"]:
    raise RuntimeError("Published fixture verification failed")
changed = copy.deepcopy(fixture["envelope"])
changed["receipt"]["economics"]["totalAuthorized"]["amount"] = "0.02"
if verify_mer(changed, **options)["valid"]:
    raise RuntimeError("Tampered fixture was accepted")
print(json.dumps({"synthetic": True, "originalVerified": True, "tamperingRejected": True, "networkRequests": 0}))
